src/term/transform/linearize.rs

```diff
@@ -111,29 +111,17 @@ fn count_var_uses_in_term(term: &Term, uses: &mut HashMap<Name, Val>) {
   }
 }
 
-/// We use a special named variable to only count the number of actual var uses on the resulting term
-fn used_var_counter(var_name: &Name) -> Name {
-  Name(format!("${var_name}$used"))
-}
-
 /// Var-declaring terms
 fn term_with_bind_to_affine(
   term: &mut Term,
   nam: &mut Option<Name>,
   var_uses: &mut HashMap<Name, Val>,
   let_bodies: &mut HashMap<Name, Term>,
 ) {
-  if let Some(nam_some) = nam {
-    if let Some(mut uses) = var_uses.get(nam_some).copied() {
+  if let Some(name) = nam {
+    if let Some(_) = var_uses.get(name).copied() {
```

Comment:
If we don't check the number of uses anymore here, you can change it to `var_uses.contains_key(name)`